rust/src/modbus/log.rs

```diff
@@ -27,7 +27,6 @@ pub extern "C" fn rs_modbus_to_json(tx: &mut ModbusTransaction, js: &mut JsonBui
 
 /// populate a json object with transactional information, for logging
 fn log(tx: &ModbusTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
-    js.open_object("modbus")?;
```

Comment:
I wouldn't say there is a right way, but we want to avoid empty objects, or even the overhead of empty objects.  So I think the caller should only open the object if it knows it will contain data.  If its the logging function that decides whether logging will happen or not, maybe it should open the object, to avoid the case where the caller would have the rewind to a marker.

Reply:
The way you put it, it looks better to open in the logging function which knows better if the object is empty